frontend/components/form/FormTextField.vue

```diff
@@ -0,0 +1,56 @@
+<script setup lang="ts">
+import useUniqueID from "~/composables/useUniqueID";
+import useFormInput from "~/composables/useFormSetup";
+
+const props = defineProps({
+  placeholder: {
+    type: String,
+    default: "",
+  },
+  modelValue: {
+    type: String,
+    default: "",
+  },
+  inputType: {
+    type: String,
+    default: "text",
+  },
+  isIconVisible: {
+    type: Boolean,
+    default: false,
+  },
+});
+const emit = defineEmits(["update:modelValue"]);
+
+
+const { updateValue } = useFormInput({ value: props?.modelValue }, emit, false);
+
+const uuid = useUniqueID().getID();
+
+const refInputType = ref(props?.inputType)
+
+const changeInputType = () => {
+  refInputType.value = refInputType.value === "password" ? "text" : "password"
+}
+
+</script>
+
+<template>
+  <div
+    class="flex items-center pl-[12px] pr-[10px] py-2 max-h-[40px] space-x-2 text-left border rounded select-none border-light-interactive dark:border-dark-interactive text-light-special-text dark:text-dark-special-text"
+  >
+    <input
+      class="w-full h-5 bg-transparent outline-none placeholder:text-light-special-text dark:placeholder:text-dark-special-text"
+      @input="updateValue"
+      :id="uuid"
+      :value="modelValue"
+      :placeholder="placeholder"
+      :type="refInputType"
+    />
+    <span v-if="isIconVisible" @click="changeInputType" class="cursor-pointer">
+      <Icon name="mdi:eye" size="28px" />
+    </span>
+  </div>
+</template>
+
+<style scoped></style>
```

Comment:
No need to add in the style block if it's not being used :) And we normally do template before script, which I know is maybe a bit against what the normal ordering is. Would be good if you could share some feedback on why that's the standard, but for me it's always easier to think of the HTML first 😊

I can remove this myself. No stress!